lib/rmt/cli/repos_custom.rb

```diff
@@ -0,0 +1,144 @@
+class RMT::CLI::ReposCustom < RMT::CLI::Base
+
+  include ::RMT::CLI::ArrayPrintable
+
+  desc 'add URL NAME', 'Creates a custom repository.'
+  def add(url, name)
+    previous_repository = Repository.find_by(external_url: url)
+
+    if previous_repository
+      warn 'A repository by this URL already exists.'
+      return
+    end
+
+    begin
+      repository_service.create_repository!(nil, url, {
+        name: name,
+        mirroring_enabled: 1,
+        autorefresh: 1,
+        enabled: 0
+      }, custom: true)
+
+      puts 'Successfully added custom repository.'
+    rescue RepositoryService::InvalidExternalUrl => e
+      warn "Invalid URL \"#{e.message}\" provided."
```

Comment:
That should raise `RMT::CLI::Error`.

Reply:
This has been updated but github doesn't seem to pick up on it.